components/inputs/docs/input-textarea.md

```diff
@@ -0,0 +1,81 @@
+# TextAreas
+
+The `<d2l-input-textarea>` is a wrapper around the native `<textarea>` element that provides auto-grow and validation behaviours. It's intended for inputting unformatted multi-line text.
+
+![example screenshot of text input](../screenshots/textarea.gif?raw=true)
+
+```html
+<script type="module">
+  import '@brightspace-ui/core/components/inputs/input-textarea.js';
+</script>
+<d2l-input-textarea
+  label="Description"
+  placeholder="Description of your topic."
+  value="Some description..."></d2l-input-textarea>
+```
+
+**Properties:**
+
+| Property | Type | Description |
+|--|--|--|
+| `aria-invalid` | String | Indicates that the `textarea` value is invalid |
+| `description` | String | A description to be added to the `textarea` for accessibility |
+| `disabled` | Boolean | Disables the `textarea` |
+| `label` | String, required | Label for the `textarea` |
+| `label-hidden` | Boolean | Hides the label visually (moves it to the `textarea`'s `aria-label` attribute) |
+| `max-height` | String, default: 12rem | Maximum height of the input before scrolling. `none` allows `textarea` to grow infinitely. |
```

Comment:
Curious whether it would be possible to just rely on the CSS `style="max-height: 5rem; min-height: 2rem;"` values instead of needing to define our own? Or is it because changing them needs to trigger a re-render?

Reply:
I can experiment with that - it might work.

Reply:
I can get the auto-grow to work properly (with the `style="min/max-height"`) by making the host `display: flex; flex-direction: column;`, the label `flex: none` and the remainder `flex: auto;` with `overflow: auto;`, and also move the `d2l-tooltip` out of the container our so it doesn't get clipped by the overflow.

However it creates some other problems: 

* the label ends up using some of the height, and the amount depends on whether the label wraps or not, so it's sensitive to browser resizing too.  Maybe we could be ok with this, but it seems weird, and if design wants a particular usage to have a certain input height, it wouldn't be guaranteed. 
* it surfaces the `display` requirement on the `host`, and if the consumer overrides it then the behaviour breaks.  If we want to allow the consumer to use `display: inline-block;` with horizontal auto-grow, then that further complicates things because they'd need to use `display: inline flex`, which so far I've been unable to get working properly with the auto-grow, but even if it worked, I don't think the consumer should need to know the internal flex requirement.

Maybe I am missing some other options, although I've played around with it longer than I'd like to admit.  😆  I'm open to exploring this further, but if the label issue is a deal breaker then I opt for not spending more time ironing out the `display` issue.

Reply:
Yeah doesn't seem worth it if it's causing this much grief -- mostly I like avoiding creating properties/attributes for things that exist already in CSS. What if instead we called it `max-cols` and `min-cols` and then internally translated those integers to `rem` units? That way it's less confusing with the CSS.

Reply:
I'm guessing you mean rows, or lines?  Maybe `min-lines` / `max-lines`?

Reply:
Right yes, rows or lines!